stdlib/math.ysh

```diff
@@ -1,79 +1,79 @@
+func _select(list, cmp) {
+  ## Internal helper for `max` and `min`.
+  ##
+  ## NOTE: If `list` is empty, then an error is thrown.
 
+  if (len(list) === 0) {
+    error ("Unexpected empty list", status=3)
+  }
 
-# We recommend abs(float1 - float2) < 0.0001
-# instead of exact float equality
+  if (len(list) === 1) {
+    return (list[0])
+  }
 
-# This causes a name collision with `abs`. The rename is a work around for now.
-func abs_(x) {
-  if (x < 0) {
-    return (-x)
-  } else {
-    return (x)
+  var match = list[0]
+  for i in (1:len(list)) {
+    setvar match = cmp(list[i], match)
```

Comment:
It's cool this higher order function works!


It reminds me that we may want a `strcmp()`, because `< >` are arithmetic only on purpose, unlike in Python